### DISCUSSION TITLE
Changing Source Labels?

### DISCUSSION THREAD

**Discussion by DeviousAl** [mvdwetering/yamaha_ynca#332] [General]
Anyone come up with a way to change / override the names / label of the sources and how they are shown in HA.

For example - AUDIO1 = Chromecast, Av4 = Shield, AV2 = Apple TV and so on?

That way when using media player card - can have it show sources to select and they are meaningful?

**Comment by DeviousAl** [comment]
For reference on the mini-media-player I'm getting around this by using shortcuts to push a source change - rather than just allowing the player to list the input options:

'''
type: custom:mini-media-player
entity: media_player.yamaha_cinema_reciever
icon: mdi:theater
hide:
  volume: false
  source: true
  power_state: false
  shortcuts: false
sound_mode: full
info: short
volume_stateless: true
volume_step: '0.5'
name: Yamaha Amp
shortcuts:
  label: Select Input
  hide_when_off: true
  list:
    - name: Shield
      type: source
      id: AV4
    - name: Music - Mobile Chromecast
'''

**Comment by mvdwetering** [comment]
If you change the input names through the menus on the receiver the integration should pick up the names. 
Only downside is that the names are limited to 9 characters so "Chromecast" does not fit 😞

Blu-ray, Cromecast, SteamLink, Shield and XBox are custom names I gave to my inputs.

![image](https://github.com/user-attachments/assets/1298e3c4-32d7-45de-ae30-6307f64fc81d)

**Comment by DeviousAl** [comment]
Interesting. Can we create a HA entity for each source then they be some customisable within HA itself?

Just thinking outloud as not sure how your hacs integration works or handles naming!

Cheers
Alex

**Comment by mvdwetering** [comment]
Technically it is possible to add a Text entity for each input that supports renaming, but that would add a lot of entities (18 for mine) which seems a bit excessive for something that rarely changes after initial setup.

I have seen some integrations having a custom configuration page which seems more suitable for something like this, but that seems like a lot of work for just renaming some inputs.

I need to think about this, but I don't think I will add something on the short term.
